common/src/lib.rs

```diff
@@ -6,34 +6,40 @@ use std::{thread::sleep, time::Duration};
 use substrate_api_client::{Api, RpcClient};
 pub use ws_rpc_client::WsRpcClient;
 
-pub trait FromStr {
-    fn from_str(s: &str) -> Self;
+pub trait FromStr: Sized {
+    type Err;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err>;
 }
 
 impl FromStr for substrate_api_client::rpc::ws_client::WsRpcClient {
-    fn from_str(url: &str) -> Self {
-        substrate_api_client::rpc::ws_client::WsRpcClient::new(url)
+    type Err = ();
+
+    fn from_str(url: &str) -> Result<Self, Self::Err> {
+        Ok(substrate_api_client::rpc::ws_client::WsRpcClient::new(url))
     }
 }
 
 impl FromStr for WsRpcClient {
-    fn from_str(url: &str) -> Self {
+    type Err = String;
+
+    fn from_str(url: &str) -> Result<Self, Self::Err> {
         WsRpcClient::new(url)
     }
 }
 
 pub fn create_connection(
     address: String,
 ) -> Api<sr25519::Pair, substrate_api_client::rpc::ws_client::WsRpcClient> {
-    create_custom_connection(&address)
+    create_custom_connection(&address).expect("connection should be created")
 }
 
 pub fn create_custom_connection<Client: FromStr + RpcClient>(
     address: &str,
-) -> Api<sr25519::Pair, Client> {
-    let client = Client::from_str(&format!("ws://{}", address));
+) -> Result<Api<sr25519::Pair, Client>, <Client as FromStr>::Err> {
```

Comment:
This function never returns `Err` it should return just `Api`

Reply:
What about line :40?

Reply:
Im blind